src/domain/studyLevel/useStudyLevels.ts

```diff
@@ -0,0 +1,47 @@
+import { StudyLevelsQuery, useStudyLevelsQuery } from '../../generated/graphql';
+import { useTranslation } from '../../i18n';
+import { translateValue } from '../../utils/translateUtils';
+
+export interface StudyLevelsState {
+  data: StudyLevelsQuery | undefined;
+  loading: boolean;
+  options: StudyLevelOption[];
+}
+
+export interface StudyLevelOption {
+  label: string;
+  value: string;
+}
+
+/**
+ * Fetch study levels from API and make options of them.
+ */
+export default function useStudyLevels(): StudyLevelsState {
+  const { t } = useTranslation();
+  const { data: studyLevelsData, loading } = useStudyLevelsQuery();
+
+  const studyLevelOptions = studyLevelsData?.studyLevels?.edges
+    ? studyLevelsData?.studyLevels?.edges?.map((studyLevelNodeEdge) => {
```

Comment:
Perhaps a bit shorter and more readable would be `studyLevelsData?.studyLevels?.edges?.map(....) ?? [];`

Reply:
I didn't know that such an option exists 👍 - changed!